stdlib/LinearAlgebra/src/diagonal.jl

```diff
@@ -675,7 +675,7 @@ for f in (:exp, :cis, :log, :sqrt,
 end
 
 function inv(D::Diagonal{T}) where T
-    Di = similar(D.diag, typeof(inv(zero(T))))
+    Di = similar(D.diag, typeof(inv(one(T))))
```

Comment:
```suggestion
    Di = similar(D.diag, typeof(inv(oneunit(T))))
```
Otherwise this might break `inv` for unitful matrices. Maybe add a test for those as well?

Reply:
Will do.

Reply:
I tried to add unitful tests, but it's way too tedious. To handle arithmetic operations for mock unitful numbers, as well as norms, quite a bit of infrastructure is necessary. It doesn't make sense to build this just for the `diagonal.jl` test case. It would make more sense to just `using Unitful`. Or to add such tests to `Unitful.jl`.